src/azul/service/source_controller.py

```diff
@@ -53,13 +61,21 @@ def _list_source_ids(self,
         return {source['sourceId'] for source in sources}
 
     def _add_implicit_sources_filter(self,
+                                     catalog: CatalogName,
                                      explicit_filters: MutableFilters,
                                      source_ids: Set[str]
                                      ) -> None:
         # We can safely ignore the `within`, `contains`, and `intersects`
         # operators since these always return empty results when used with
         # string fields.
-        explicit_source_ids = explicit_filters.setdefault('sourceId', {}).get('is')
-        if explicit_source_ids is not None:
-            source_ids = source_ids.intersection(explicit_source_ids)
-        explicit_filters['sourceId']['is'] = list(source_ids)
+        source_id_facet = self.service.service_config(catalog).source_id_facet
+        facet_filter = explicit_filters.setdefault(source_id_facet, {})
+        try:
+            is_filter = facet_filter['is']
+        except KeyError:
+            facet_filter['is'] = list(source_ids)
+        else:
+            inaccessible = set(is_filter) - source_ids
+            if inaccessible:
+                raise ForbiddenError(msg=f'Cannot filter by inaccessible sources: {source_ids}')
+        assert set(explicit_filters[source_id_facet]['is']) <= source_ids
```

Comment:
This assert seems unnecessary, isn't this already guaranteed by the try / except / else?